packages/format-library/src/link/utils.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Check for issues with the provided link URL.
+ * (e.g. invalid characters, or an unexpected syntax).
+ *
+ * @param {string} linkURL The link URL.
+ *
+ * @return {boolean} Is the URL invalid?
+ */
+export function isInvalidLinkURL( linkURL ) {
+	const trimmedURL = linkURL.trim();
```

Comment:
Here's the existing validation for comparison:
https://github.com/WordPress/WordPress/blob/da7a80d67fea29c2badfc538bfc01c8a585f0cbe/wp-includes/js/tinymce/plugins/wplink/plugin.js#L154-L171